core/service/rollout.go

```diff
@@ -0,0 +1,290 @@
+	groupRolloutRelationDao   dao.GroupRolloutRelation
+	appRolloutRelationDao     dao.AppRolloutRelation
+	rolloutVersionRelationDao dao.RolloutVersionRelation
+	timeRangeActivatorDao     dao.TimeRangeActivator
+	maxViewersActivatorDao    dao.MaxViewersActivator
+	percentageActivatorDao    dao.PercentageActivator
+	appVersionDao             dao.AppVersion
+}
+
+func (r *Rollout) FindUserRolloutsByAppID(ct context.Context, appID uint64) ([]entity.Rollout, *errs.Error) {
+	rolloutIDs, err := r.appRolloutRelationDao.FindRolloutIDsByAppIDAndType(ct, appID, entity.AppRolloutRelationTypeUser)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.rolloutDao.FindRolloutsByIDs(ct, rolloutIDs)
+}
+
+func (r *Rollout) FindTeamRolloutsByAppID(ct context.Context, appID uint64) ([]entity.Rollout, *errs.Error) {
+	rolloutIDs, err := r.appRolloutRelationDao.FindRolloutIDsByAppIDAndType(ct, appID, entity.AppRolloutRelationTypeTeam)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.rolloutDao.FindRolloutsByIDs(ct, rolloutIDs)
+}
+
+func (r *Rollout) CreateRollout(
+	ct context.Context,
+	selectorType entity.SelectorType,
+	activatorID uint64,
+	activatorType entity.ActivatorType,
+	isEnabled bool,
+) (entity.Rollout, *errs.Error) {
+	genSelectorIDReq := &proto.GenerateUniqueNumberRequest{SequenceName: "selectorID"}
+	genSelectorIDRes, rpcErr := r.cloudClientRegistry.GeneratorClient().GenerateUniqueNumber(ct, genSelectorIDReq)
+	if rpcErr != nil {
+		internalErr := errs.FromGRPCErr(rpcErr)
+		return entity.Rollout{}, internalErr
+	}
+
+	rollout := entity.Rollout{
+		SelectorID:    genSelectorIDRes.UniqueNumber,
+		SelectorType:  selectorType,
+		ActivatorID:   activatorID,
+		ActivatorType: activatorType,
+		IsEnabled:     isEnabled,
+	}
+
+	return r.rolloutDao.CreateRollout(ct, rollout)
+}
+
+func (r *Rollout) DeleteRollout(ct context.Context, rolloutID uint64) (entity.Rollout, *errs.Error) {
+	var rollout entity.Rollout
+	txCtx := TransactionsContext{
+		logger:             r.logger,
+		transactionFactory: r.transactionFactory,
+		stateSyncer:        r.stateSyncer,
+		ct:                 ct,
+	}
+
+	err := txCtx.withTransactions(false, func(tx *transaction.Transaction, rtTx *realtime.Transaction) *errs.Error {
+		var err *errs.Error
+		rollout, err = r.rolloutDao.FindRolloutByIDWithTx(ct, tx, rolloutID)
+		if err != nil {
+			return err
+		}
+
+		return r.rolloutDao.DeleteRolloutWithTx(ct, tx, rolloutID)
+	})
+
+	if err != nil {
+		return entity.Rollout{}, err
+	}
+
+	return rollout, nil
+
+}
+
+func (r *Rollout) FindRolloutsByGroupID(ct context.Context, groupID uint64) ([]entity.Rollout, *errs.Error) {
+	rolloutIDs, err := r.groupRolloutRelationDao.FindRolloutIDsByGroupIDAndSortByOrderedIndex(ct, groupID)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.rolloutDao.FindRolloutsByIDs(ct, rolloutIDs)
+}
+
+func (r *Rollout) GetActiveAppVersionNumberForTeam(ct context.Context, appTeamInstallation entity.TeamAppInstallation) (int, *errs.Error) {
```

Comment:
```suggestion
func (r *Rollout) GetActiveAppVersionNumberForTeam(ct context.Context, appTeamInstallation entity.TeamAppInstallation) (*int, *errs.Error) {
```

ActiveVersion is `nil` when there is no rollout.